src/models/wishDao.js

```diff
@@ -0,0 +1,52 @@
+const { dataSource } = require('./dataSource');
+const queryBuilder = require('./queryBuilder');
+
+const postWish = async (userId, campId) => {
+  try {
+    const existWish = await dataSource.query(
+      `SELECT
+      user_id,
+      camp_id
+      FROM wishlists
+      WHERE user_id = ? AND camp_id = ?`,
+      [userId, campId]
+    );
+
+    if (existWish.length === 0) {
+      await dataSource.query(
+        `INSERT INTO wishlists (
+          user_id,camp_id
+          )VALUES (?,?)`,
+        [userId, campId]
+      );
+      return await dataSource.query(
+        `SELECT 
+              camp_id
+              FROM wishlists
+              WHERE user_id = ? AND camp_id =?`,
+        [userId, campId]
+      );
+    } else {
+      await dataSource.query(
+        `DELETE FROM wishlists
+        WHERE user_id =? AND camp_id =?`,
+        [userId, campId]
+      );
+      return await dataSource.query(
+        `SELECT 
+              camp_id
+              FROM wishlists
+              WHERE user_id = ? AND camp_id =?`,
+        [userId, campId]
+      );
+    }
```

Comment:
```suggestion
const postWish = async (userId, campId) => {
  try {
    await dataSource.query(
      `INSERT INTO wishlists (user_id, camp_id)
      VALUES (?, ?)
      ON DUPLICATE KEY DELETE`,
      [userId, campId]
    );

    const result = await dataSource.query(
      `SELECT camp_id FROM wishlists WHERE user_id = ? AND camp_id = ?`,
      [userId, campId]
    );

    return result;
  } catch (err) {
    throw new Error('INVALID_DATA_INPUT 😮');
  }
};
```

- 위 내용을 혹시 의도한 것일까요?

Reply:
- 수정을 한 번 고민해보세요!

Reply:
getOrUpdateWish.affectedRows==0 
에서 .affectedRows :  데이터베이스 작업에서 영향을 받은 행의 수를 나타내는 것 !
== 0 이란건 행의수가 없는것 !